stdlib/public/core/Misc.swift

```diff
@@ -132,3 +132,18 @@ public func _getTypeByMangledNameInContext(
   genericContext: UnsafeRawPointer?,
   genericArguments: UnsafeRawPointer?)
   -> Any.Type?
+
+@_silgen_name("swift_getMetadataSection")
+public func _getMetadataSection(
+        _ index: UInt)
+                -> UnsafeRawPointer?
+
+@_silgen_name("swift_getMetadataSectionCount")
+public func _getMetadataSectionCount()
+                -> UInt
+
+@_silgen_name("swift_getMetadataSectionName")
+public func _getMetadataSectionName(
+        _ metadata_section: UnsafeRawPointer)
+                -> UnsafePointer<CChar>
```

Comment:
These don't seem to be following the standard library style.

Reply:
Are these declarations conditionalized to when we're building a debug runtime that contains the symbols?

Reply:
@jckarter I just remembered why they aren't conditionalized here. The problem is that `SwiftReflectionTest.swift` is built even when assertions are off. To conditionalize them here, I'd have to change the CMake config to only build this file in debug or on Mac. I initally did that, but was asked to roll-back that change. These functions are conditionalized in C++ though, so I believe calling them would crash the program, which isn't ideal. 
In order to conditionalize these symbols, in `SwiftReflectionTest.swift`, when on Linux but not in debug, stub the required functions so it as least compiles. Is this solution ok?

Reply:
What was wrong with conditionalizing SwiftReflectionTest.swift? If touching CMake is inappropriate, maybe we could #if out the contents.